base/iobuffer.jl

```diff
@@ -110,15 +215,13 @@ function IOBuffer(
         append::Union{Bool,Nothing}=nothing,
         truncate::Union{Bool,Nothing}=nothing,
         maxsize::Integer=typemax(Int),
-        sizehint::Union{Integer,Nothing}=nothing)
-    if maxsize < 0
-        throw(ArgumentError("negative maxsize"))
-    end
+        sizehint::Union{Integer,Nothing}=nothing,
+    )
```

Comment:
Probably it's better to keep the diff smaller by avoiding unrelated stylistic changes?

```suggestion
        sizehint::Union{Integer,Nothing}=nothing)
```

Reply:
Thanks for the review - this and the other comment of you have been addressed as you suggested